File: CFS/80A.cpp
#include <bits/stdc++.h>
using namespace std;
#define endl '\n'
#define vi vector<int>

const int N = 51;
vector<bool> p(N);
vi primes;

void sieve()
{
  for (int i = 0; i < N; i++)
    p[i] = true;

  p[0] = p[1] = false;

  for (int i = 2; i * i < N; i++)
    if (p[i])
      for (int j = i * i; j < N; j += i)
        p[j] = false;

  for (int i = 0; i < N; i++)
    if (p[i])
      primes.push_back(i);
}

void solve(int now, int next)
{
  for (int i = 0; i < primes.size(); i++)
  {
    if (primes[i] == now)
    {
      if (i + 1 < primes.size() && primes[i + 1] == next)
        cout << "YES\n";
      else
        cout << "NO\n";
      return;
    }
  }
  cout << "NO\n";
}

int main()
{
  int now, next; cin >> now >> next;
  sieve();
  solve(now, next);
}

File: CFS/80A.py
import math
N=100
p=[True]*(N+1)
p[0]=p[1]=False
primes=[]

for i in range(2,int(math.sqrt(N))):
  if p[i]:
    for j in range(i*i,N,i):
      p[j]=False
for i in range(N):
  if p[i]:
    primes.append(i)

def solve(now,next):
  for i in range(0,len(primes)):
    if primes[i]==now:
      if i+1<len(primes) and primes[i+1]==next:
        print('YES')
      else:
        print('NO')
      return
  print('NO')

now,next=map(int,input().split())
solve(now,next)